src/ray/object_manager/plasma/store.cc

```diff
@@ -321,6 +323,21 @@ PlasmaError PlasmaStore::HandleCreateObjectRequest(const std::shared_ptr<Client>
     RAY_LOG(DEBUG) << "Not enough memory to create the object " << object_id
                    << ", data_size=" << data_size << ", metadata_size=" << metadata_size;
   }
+
+  // Trigger object spilling if current usage is above the specified threshold.
+  if (spilling_required != nullptr) {
+    const int64_t footprint_limit = PlasmaAllocator::GetFootprintLimit();
+    if (footprint_limit != 0) {
+      const float allocated_percentage =
+          ((float)PlasmaAllocator::Allocated()) / footprint_limit;
```

Comment:
Nit: Should prefer explicit `static_cast` over C-style casts.
```suggestion
          static_cast<float>(PlasmaAllocator::Allocated()) / footprint_limit;
```